rules/fusesoc.bzl

```
@@ -36,23 +40,32 @@ def _fusesoc_build_impl(ctx):
         verilator_options = ctx.attr.verilator_options[BuildSettingInfo].value
         flags.append("--verilator_options={}".format(" ".join(verilator_options)))
 
+    args.add_all(
+        ctx.files.cores,
+        uniquify = True,
+        map_each = _corefiles2rootarg,
```

```suggestion
        map_each = lambda core: core.dirname,
```
Not sure if this is allowed in starlark

---

It's not